server-protocol.md

```diff
@@ -104,6 +114,46 @@ was sent in the wrong order), an `error` response will be sent, This response
 will include the error string in the `error` key, and a full copy of the
 original message dictionary in `orig`.
 
+
+## Permission and Abilities
+
+Server operators may wish to deny service to some clients.
+
+One such use-case is if the server is under a Denial of Service (DoS)
+attack or other malicious activity.
+
+In the `welcome` message the server may include a `permission-required` key. If provided, this will point to a `dict` with one key per supported method along with any method-specific options.
+For example::
+
+    {
+        "none": {}
```

Comment:
Couldn't we just say that "none" is implied by omitting the `permission-required` field?

Reply:
The reason I included it here was so servers can explicitly say "I don't support `none` right now" (by not including it). But, I guess the server has to handle that with an error for old clients anyway, so maybe we could leave it out.

Reply:
Here's a concrete case: if a client connects and only sees "`hashcash`" in the list then it's hard to decide to send nothing. That could result in an error (if the server is disallowing "nothing" connections) and the client would have to try again..

Whereas, if it sees "`none`" and "`hashcash`" in the list, the server is signalling (to new clients) that it will accept a `bind` with no `submit-permission`  key.

Reply:
Hm yeah, maybe we should talk about the default behavior for the public Wormhole rendezvous server. I'd like to have all clients to always use `hashcash`, albeit with a trivially low number of bits when not under load. And of course it'll usually accept no hashcash. But if the server actively advertizes that it also supports `none`, then why would clients bother doing the work?

Reply:
Right, the point is if the server isn't demanding hashcash currently then the client can save work and not submit it.

That's up to the server operator, but at the very least I'd expect "the" server will want a transition-period where it _supports_ accepting hashcash but "usually" doesn't demand it .. this allows for a slower upgrade of the client ecosystem. If the server started demanding hashcash always then old clients simply won't work.